hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/HapiThrottling.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.throttle;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.transaction.Query;
+import com.hedera.node.app.state.HederaState;
+import com.hedera.node.app.workflows.TransactionInfo;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import javax.inject.Inject;
+
+public class HapiThrottling {
+
+    private final GeneralThrottleAccumulator generalThrottleAccumulator;
+
+    @Inject
+    public HapiThrottling(GeneralThrottleAccumulator generalThrottleAccumulator) {
+        this.generalThrottleAccumulator =
+                requireNonNull(generalThrottleAccumulator, "generalThrottleAccumulator must not be null");
+    }
+
+    public synchronized boolean shouldThrottle(@NonNull TransactionInfo txnInfo, HederaState state) {
```

Comment:
Sorry, I did not realize this when reviewing `HandleThrottleAccumulator`. I think we should make the names of the methods consistent. Either both methods are called `shouldThrotte()` or we call them `shouldThrottleTransaction()` and `shouldThrottleQuery()`. I'd prefer the first but have no strong opinion.